controllers/devworkspace/solver/endpoint_exposer.go

```diff
@@ -239,18 +240,24 @@ func (e *IngressExposer) getIngressForService(endpoint *EndpointInfo, endpointSt
 	return ingress
 }
 
-func routeAnnotations(machineName string, endpointName string) map[string]string {
-	return map[string]string{
-		defaults.ConfigAnnotationEndpointName:  endpointName,
-		defaults.ConfigAnnotationComponentName: machineName,
+func routeAnnotations(endpointAnnotations map[string]string, machineName string, endpointName string) map[string]string {
+	annos := map[string]string{}
+	for k, v := range endpointAnnotations {
+		annos[k] = v
 	}
+	annos[defaults.ConfigAnnotationEndpointName] = endpointName
+	annos[defaults.ConfigAnnotationComponentName] = machineName
+	return annos
 }
 
-func finalizeIngressAnnotations(ingressAnnotations map[string]string, machineName string, endpointName string) map[string]string {
+func finalizeIngressAnnotations(ingressAnnotations, endpointAnnotations map[string]string, machineName string, endpointName string) map[string]string {
 	annos := map[string]string{}
 	for k, v := range ingressAnnotations {
 		annos[k] = v
 	}
+	for k, v := range endpointAnnotations {
+		annos[k] = v
```

Comment:
**Note:** the devfile endpoint annotations can actually override the values of the [Che Cluster CR ingress annotations](https://github.com/eclipse-che/che-operator/blob/main/api/v2/checluster_types.go#L238-L245). To me, this makes sense, as it gives users more fine-grained control over the ingress annotations. However, I don't have a strong opinion on this.

Reply:
I realized we are not adding annotations from the Che Cluster CR's `networking.annotations` fields to the routes created by the Che Router, only the ingresses get these annotations.

This seems like a bug or an error in the documentation, so I've opened https://github.com/eclipse-che/che/issues/23118 to track this. If this is indeed a bug, I could add the fix as part of this PR since it's closely related.